src/mdb/utils.py

```diff
@@ -34,11 +34,15 @@ def extract_float(line: str, backend: "DebugBackend") -> float:
     float_regex = backend.float_regex
     line = strip_control_characters(line)
     m = re.search(float_regex, line)
+
     if m:
         try:
             result = float(m.group(1))
         except ValueError:
             print(f"cannot convert variable [{result}] to a float.")
+    
+    # Corner case: if result is uninitialized, the parent function in the callstack
+    # should capture the fired exception of UnboundLocalError.
```

Comment:
```suggestion
    if m:
        try:
            result = float(m.group(1))
        except ValueError:
            raise ValueError(f"cannot convert variable [{result}] to a float.")
    else:
        msg = "float regex doesn't match following string:\n"
        msg += line
        raise ValueError(msg)
```

Thanks @Accumulated . I would prefer this approach. I think it's best not to return `nan` but instead raise a `ValueError` for any invalid value.

Reply:
Plus the above suggestion also prints the string to the `mdb` shell which allows the user to hopefully understand why `do_plot` cannot print it.

Reply:
Thanks @TomMelt for your suggestion. I updated the file to match the requested changes.